File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left_pointer = 0
        right_pointer = len(nums) - 1

        while left_pointer < right_pointer:
            
            mid_pointer = left_pointer + (right_pointer - left_pointer) // 2

            if nums[mid_pointer] < nums[right_pointer]:
                right_pointer = mid_pointer
            else:
                left_pointer = mid_pointer + 1
            
        
        return nums[left_pointer]

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int leftPointer = 0;
        int rightPointer = nums.length - 1 ;
        int lowest = nums[0];
        int midPointer;

        while (leftPointer < rightPointer) {
            midPointer = leftPointer + (int) Math.floor((rightPointer - leftPointer) / 2);
            if (nums[midPointer] > nums[rightPointer]) {
                leftPointer = midPointer + 1;
            }
            else {
                rightPointer = midPointer;
            }

            if (nums[midPointer] < lowest) {
                lowest = nums[midPointer];
            }            

        }

        return nums[leftPointer];
    }
}